modules/dnn/src/onnx/onnx_graph_simplifier.cpp

```diff
@@ -290,9 +294,77 @@ class ResizeSubgraph2 : public ExtractScalesSubgraph
     }
 };
 
+class UpsampleTwoInputSubgraph : public Subgraph
+{
+public:
+    UpsampleTwoInputSubgraph()
+    {
+        int input1 = addNodeToMatch("");
+        int shape = addNodeToMatch("Shape", input1);
+
+        int input2 = addNodeToMatch("");
+
+        int shape1 = addNodeToMatch("Shape", input2);
+        int indexH = addNodeToMatch("Constant");
+        int gather1 = addNodeToMatch("Gather", shape1, indexH);
+
+        int shape2 = addNodeToMatch("Shape", input2);
+        int indexW = addNodeToMatch("Constant");
+        int gather2 = addNodeToMatch("Gather", shape2, indexW);
+
+        int unsqueeze1 = addNodeToMatch("Unsqueeze", gather1);
+        int unsqueeze2 = addNodeToMatch("Unsqueeze", gather2);
+        int concatId = addNodeToMatch("Concat", unsqueeze1, unsqueeze2);
+
+        int constant = addNodeToMatch("Constant");
+        int cast1 = addNodeToMatch("Cast", concatId);
+
+        int slice = addNodeToMatch("Slice", shape);
+        int cast2 = addNodeToMatch("Cast", slice);
+
+        int div = addNodeToMatch("Div", cast1, cast2);
+        int concat = addNodeToMatch("Concat", constant, div);
+
+        addNodeToMatch("Upsample", input1, concat);
+        setFusedNode("Upsample", input1, input2);
+    }
```

Comment:
Can we inherit it from `ExtractScalesSubgraph`?

Reply:
I think no. this subgraph is quite different than `ExtractScalesSubgraph`. The main thiing is scales are not constant input of `Mul` nodes unlike `ExtractScalesSubgraph`.

Reply:
subgraph fusion for that subgraph is removed. resolving conversation.